### PR TITLE
Fix llvm for BPF builds

### PR DESCRIPTION
**Pull request by kanikanema**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 
Commit 5b511cf removed BPF from list of target-to-build, re-add it.
Upgrade xdp-tools to 1.4.2 that officially supports clang 17.x

###### Change Log  
xdp-tools build recently started failing with the following error:

time="2024-02-05T04:34:56Z" level=debug msg="/usr/bin/llc -march=bpf -filetype=obj -o xsk_def_xdp_prog_5.3.o xsk_def_xdp_prog_5.3.ll"
time="2024-02-05T04:34:56Z" level=debug msg="/usr/bin/llc: error: invalid target 'bpf'."
time="2024-02-05T04:34:56Z" level=debug msg="make[2]: *** [Makefile:139: xsk_def_xdp_prog_5.3.o] Error 1"

Commit 5b511cf removed BPF from list of target-to-build.
This breaks packages like xdp-tools that use BPF targets.
This patch re-adds the support for BPF in llvm.

For good measure upgrade xdp-tools from 1.4.1 -> 1.4.2 (latest) that includes support for clang-17.

###### Does this affect the toolchain?  
**NO**

###### Test Methodology
BuddyBuild (with tests): https://dev.azure.com/mariner-org/mariner/_build/results?buildId=498909&view=results

### CODE REVIEW

**Comment by mfrw** [comment]
Related PR:
- https://github.com/microsoft/CBL-Mariner/pull/7589

File: SPECS/llvm/llvm.spec [hunk 1]
```diff
@@ -49,7 +49,7 @@ cmake -G Ninja                              \
       -DLLVM_LINK_LLVM_DYLIB=ON             \
       -DLLVM_INCLUDE_TESTS=ON               \
       -DLLVM_BUILD_TESTS=ON                 \
-      -DLLVM_TARGETS_TO_BUILD="host"        \
+      -DLLVM_TARGETS_TO_BUILD="host;BPF"    \
```
Comment (mfrw) [inline SPECS/llvm/llvm.spec:52]: Should we have `all` here instead of just `host;BPF` ?

**Review by mfrw — APPROVED** [review]
LGTM :)

Have a clarification question for @anphel31 / @nicogbg

File: SPECS/llvm/llvm.spec (hunk 1 above)
Comment (anphel31) [reply SPECS/llvm/llvm.spec:52]: In Mariner 1.0 and 2.0 we had: `host;AMDGPU;BPF`. I think we should stick with the same until we have a reason to change this.

**Review by anphel31 — APPROVED** [review]
approving, although I think we should restore LLVM_TARGETS_TO_BUILD to "`host;AMDGPU;BPF`", which is what we've always used

File: SPECS/llvm/llvm.spec (hunk 1 above)
Comment (kanikanema) [reply SPECS/llvm/llvm.spec:52]: I am assuming AMDGPU and BPF were dropped to fix some ptests failures. I wasn't sure so I only added back BPF.

**Pull request merged**